src/main/java/org/openrewrite/java/migrate/javax/AutoGeneratedValue.java

```diff
@@ -0,0 +1,86 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.javax;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.java.*;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AutoGeneratedValue extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Attributes with automatically generated values require configuration.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds missing configuration for attributes with automatically generated values.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+
+        return Preconditions.check(new UsesType<>("javax.persistence.GeneratedValue", false), new JavaIsoVisitor<ExecutionContext>() {
+            private final AnnotationMatcher GENERATED_VALUE = new AnnotationMatcher("@javax.persistence.GeneratedValue");
+            private final AnnotationMatcher GENERATED_VALUE_AUTO = new AnnotationMatcher("@javax.persistence.GeneratedValue(strategy=javax.persistence.GenerationType.AUTO)");
+
+            @Override
+            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext ctx) {
+                Set<J.Annotation> generatedValueAnnotations = FindAnnotations.find(multiVariable, "@javax.persistence.GeneratedValue");
+                boolean addAnnotation = false;
+                if (!generatedValueAnnotations.isEmpty()) {
+                    J.Annotation generatedValueAnnotation = generatedValueAnnotations.iterator().next();
+                    List<Expression> args = generatedValueAnnotation.getArguments();
+                    addAnnotation = args == null || args.isEmpty() || GENERATED_VALUE_AUTO.matches(generatedValueAnnotation);
+                }
+
+                if (addAnnotation) {
+                    J.VariableDeclarations updatedVariable = JavaTemplate.apply(
+                            "@javax.persistence.TableGenerator(name = \"OPENJPA_SEQUENCE_TABLE\", table = \"OPENJPA_SEQUENCE_TABLE\", pkColumnName = \"ID\", valueColumnName = \"SEQUENCE_VALUE\", pkColumnValue = \"0\")",
+                            getCursor(),
+                            multiVariable.getCoordinates().addAnnotation(Comparator.comparing(
+                                    J.Annotation::getSimpleName)
+                            )
+                    );
+                    return super.visitVariableDeclarations(updatedVariable, ctx);
+                }
+                return multiVariable;
```

Comment:
Instead of two if statements that set and check a boolean, I'd rather return early where we can such that logical branches for rejected cases are immediately out of mind when reading the code.
```suggestion
                if (generatedValueAnnotations.isEmpty()) {
                    return multiVariable;
                }

                J.Annotation generatedValueAnnotation = generatedValueAnnotations.iterator().next();
                List<Expression> args = generatedValueAnnotation.getArguments();
                if (!(args == null || args.isEmpty() || GENERATED_VALUE_AUTO.matches(generatedValueAnnotation))) {
                    return multiVariable;
                }

                J.VariableDeclarations updatedVariable = JavaTemplate.apply(
                        "@javax.persistence.TableGenerator(name = \"OPENJPA_SEQUENCE_TABLE\", table = \"OPENJPA_SEQUENCE_TABLE\", pkColumnName = \"ID\", valueColumnName = \"SEQUENCE_VALUE\", pkColumnValue = \"0\")",
                        getCursor(),
                        multiVariable.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName)));
                return super.visitVariableDeclarations(updatedVariable, ctx);
```